armi/reactor/converters/uniformMesh_V2_beta.py

```diff
@@ -0,0 +1,450 @@
+            b.p.assemNum = uniAssem.p.assemNum
+            b.name = b.makeName(uniAssem.p.assemNum, i)
+            uniAssem.add(b)
+            zLower = zUpper
+
+        uniAssem.reestablishBlockOrder()
+        uniAssem.calculateZCoords()
+
+        return uniAssem
+
+    def checkForSmearing(self, aList: list, printLikeBlkSmears: bool):
+        """indicate smearing by presence of block bounds that are not aligned with uniformMesh
+
+        Parameters
+        ----------
+        aList : list, :py:class:`Assembly <armi.reactor.assemblies.Assembly>` object
+            List of assemblies which uniformMesh would be applied; will or will not have inter-block smearing\
+        printLikeBlkSmears : boolean
+            There may be inter-block smearing between similar/like blocks. E.g., fuel to fuel, plenum to plenum, etc.
+            These smearing instances may be less important than dissimilar blocks (e.g., control to plenum,
+            shield to fuel, fuel to plenum). This boolean controls whether or not the similar/like block
+            smearing cases are printed.
+        """
+        for a in aList:
+            runLog.debug(
+                "{0}...".format(a)  # pylint: disable=logging-format-interpolation
+            )
+            for i, _z in enumerate(self.uniformMesh):
+                if i == 0:
+                    bottom = 0.0
+                else:
+                    bottom = self.uniformMesh[i - 1]
+                top = self.uniformMesh[i]
+                overlap = a.getBlocksBetweenElevations(bottom, top)
+                if len(overlap) > 1:
+                    diffMaterialInBlocks = len(set(b.p.flags for b, _h in overlap)) > 1
+                    if printLikeBlkSmears or diffMaterialInBlocks:
+                        runLog.debug(
+                            "    uniform mesh block {0}, ({1:.3f},{2:.3f})".format(
+                                i, bottom, top
+                            )
+                        )
+                        runLog.debug("    percent contribution \t block")
+                        for b, val in overlap:
+                            runLog.debug(
+                                "    {0:9.2f}\t\t\t{1}".format(
+                                    val / (top - bottom) * 100.0, b.p.flags
+                                )
+                            )
+
+
+def createNewBlock(overlappingBlockInfo: list, uniformHeight: float):
+    """create new armi block from overlapping block info"""
+    # get source block type
+    heights = [h for _b, h in overlappingBlockInfo]
+    blocks = [b for b, _h in overlappingBlockInfo]
+    sourceBlk = blocks[heights.index(max(heights))]
+
+    # create a new block, b, that is the same type and contains the same params, grid, etc as sourceBlk
+    b = sourceBlk.__class__(sourceBlk.getType())
+    for param in sourceBlk.getParamNames():
+        b.p[param] = sourceBlk.p[param]
+    for c in sourceBlk.getChildren():
+        b.add(c.__copy__())
```

Comment:
@john-science Just wanted to bring this section to your attention since you have the open issue, https://github.com/terrapower/armi/issues/706, on deepcopies. 

I made a concerted effort to avoid using `deepcopy` when creating a copy of an assembly -> block -> component, but gave in with components. I decided to use the `Component::__copy__` method with the hope that when #706 is resolved, it too will be resolved. Anyway, just figured I'd get your eyes on this section in particular.

Reply:
Thanks for keeping it in mind.

My concern is that "fixing" that will break a dozen major plugins, so it won't be possible for months.

Fingers crossed I'm just paranoid.  We'll find out!